File: io-hotmoka-remote/src/main/java/io/hotmoka/remote/internal/websockets/client/stomp/StompHeaders.java
package io.hotmoka.remote.internal.websockets.client.stomp;

import java.util.HashMap;
import java.util.Map;
import java.util.NoSuchElementException;

/**
 * A simple class to represent the STOMP headers.
 */
public class StompHeaders {
    private final Map<String, String> headers = new HashMap<>();

    public void add(String key, String value) {
        headers.put(key, value);
    }

    public String getDestination() {
        String destination = headers.get("destination");
        if (destination == null)
            throw new NoSuchElementException("Destination not found");

        return destination;
    }

    @Override
    public String toString() {
        return "StompHeaders{headers=" + headers + '}';
    }
}

File: io-hotmoka-network-thin-client/io-hotmoka-network-thin-client-kt/src/main/kotlin/io/hotmoka/network/thin/client/webSockets/stomp/StompHeaders.kt
package io.hotmoka.network.thin.client.webSockets.stomp

import java.util.NoSuchElementException
import kotlin.jvm.Throws

/**
 * A simple class which represents the STOMP headers.
 */
class StompHeaders {
    private val headers: MutableMap<String, String> = mutableMapOf()

    fun add(key: String, value: String) {
        headers[key] = value
    }

    private fun getHeader(key: String): String? {
        return headers[key]
    }

    @Throws(NoSuchElementException::class)
    fun getDestination(): String {
        return getHeader("destination") ?: throw NoSuchElementException("Destination not found")
    }

    override fun toString(): String {
        return headers.toString()
    }
}